### PR TITLE
[ENHANCEMENT] Allow to define the overriding configuration persist level

### PR DESCRIPTION
**Pull request by crouillon**
Introduce two optional configuration parameters to Config:config section:
 * _persist_per_context_: persist overriding config taking account of the context (true by default)
 * _persist_per_environment_: persist overriding config taking account of the environment (true by default)

### CODE REVIEW

File: Config/Persistor/File.php [hunk 1]
```diff
@@ -75,12 +91,12 @@ public function persist(Config $config, array $configToPersist)
     private function getConfigDumpRightDirectory($baseDir)
     {
         $configDumpDir = $this->app->getRepository();
```
Comment (eric-chau) [inline Config/Persistor/File.php]: Shouldn't we use `::getBaseRepository()` instead? or just use `getRepository()` so you don't need to care anymore about environment?

**Comment by eric-chau** [comment]
great PR @crouillon. Thank you :+1: just take a look at my comment ;)

File: Config/Persistor/File.php (hunk 1 above)
Comment (crouillon) [inline Config/Persistor/File.php]: In fact, I think that we must take care of the context.

Till now the configuration persisting (File or Registry) was saved on the more precise level (ie taking care of context and environment).

I think it's always a good practice by default, but in some cases being able to save it on higher level can be useful ; example: the visitor website use a different environment than the contributor website but the bundle overriding config has to be shared => _persist_per_environment: false_ allows that now.

Comment (crouillon) [inline Config/Persistor/File.php]: At last, thanks @eric-chau you're right, getBaseRepsitory() has to be used

Comment (eric-chau) [inline Config/Persistor/File.php]: Cool then I will merge it when travis-ci is full green =)

**Pull request merged**